### PR TITLE
OCPBUGS-33903: Simplify output when not updating

### PR DESCRIPTION
**Pull request by petr-muller**
Specific condition details are unlikely to be useful:

```console
$ oc adm upgrade status
The cluster version is not updating (Progressing=False).

  Reason: <none>
  Message: Cluster version is 4.14.1
```

Sometimes they can get even confusing, especially when the command picks up CVO `Failing=True` condition present before the update attempt:

```console
$ oc adm upgrade status --details=none
The cluster version is not updating (Progressing=False).

  Reason: MultipleErrors
  Message: Error while reconciling 4.16.0-0.nightly-2024-05-07-025557: an unknown error has occurred: MultipleErrors
```

I also changed one user-facing occurrence of "upgrade" to "update" as a rider in this PR

### CODE REVIEW

**Comment by petr-muller** [comment]
I realized previous output could have been useful when the user would be expecting an update but it does not start for some reason: prechecks, download, verification. Maybe it would be useful if we conditionally emited something if spec desired differs from status desired?

File: pkg/cli/admin/upgrade/status/status.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func (o *options) Run(ctx context.Context) error {
 		if message = progressing.Message; message == "" {
 			message = "<none>"
 		}
-		fmt.Fprintf(o.Out, "The cluster version is not updating (%s=%s).\n\n  Reason: %s\n  Message: %s\n", progressing.Type, progressing.Status, reason, strings.ReplaceAll(message, "\n", "\n  "))
+		fmt.Fprintf(o.Out, "The cluster is not updating.")
```
Comment (Davoska) [inline pkg/cli/admin/upgrade/status/status.go:264]: We should include a newline in the output IMO.

```suggestion
		fmt.Fprintf(o.Out, "The cluster is not updating.\n")
```

Comment (Davoska) [inline pkg/cli/admin/upgrade/status/status.go:264]: We can remove the logic regarding the `reason` and `message` variables in this condition as they are not used.

```go
	if progressing.Status != configv1.ConditionTrue && !isWorkerPoolOutdated {
		fmt.Fprintf(o.Out, "The cluster is not updating.\n")
		return nil
	}
```

**Comment by Davoska** [comment]
> I realized previous output could have been useful when the user would be expecting an update but it does not start for some reason: prechecks, download, verification. 

That is true. 

> Maybe it would be useful if we conditionally emited something if spec desired differs from status desired?

I do like the idea of this. We could work on this in the future. If you think implementation-wise this is easy to implement and you are interested in this little improvement, feel free to include it in the PR. Otherwise, we can follow up on this.

File: pkg/cli/admin/upgrade/status/status.go (hunk 1 above)
Comment (evakhoni) [reply pkg/cli/admin/upgrade/status/status.go:264]: correct. just verified it prints % in stdout

**Comment by evakhoni** [comment]
> > I realized previous output could have been useful when the user would be expecting an update but it does not start for some reason: prechecks, download, verification.
> 
> That is true.
> 
> > Maybe it would be useful if we conditionally emited something if spec desired differs from status desired?
> 
> I do like the idea of this. We could work on this in the future. If you think implementation-wise this is easy to implement and you are interested in this little improvement, feel free to include it in the PR. Otherwise, we can follow up on this.

imo can be a separate card in the future, maybe something a bit more detailed on why do we're unable to start the upgrade, and ideally what can we do about it :thinking:

File: pkg/cli/admin/upgrade/status/status.go (hunk 1 above)
Comment (petr-muller) [reply pkg/cli/admin/upgrade/status/status.go]: Addressed :+1:

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**